kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -151,117 +156,109 @@ function redirectToExploreChannel(store) {
-      const childrenPromise = ContentNodeResource.getCollection({ parent: rootTopicId }).fetch();
-      _updateChannelList(store);
-
-      ConditionalPromise.all([attributesPromise, childrenPromise]).only(
-        samePageCheckGenerator(store),
-        ([attributes, children]) => {
-          const pageState = { rootTopicId };
-          pageState.topic = _topicState(attributes);
-          const collection = _collectionState(children);
-          pageState.subtopics = collection.topics;
-          pageState.contents = collection.contents;
-          store.dispatch('SET_PAGE_STATE', pageState);
-          store.dispatch('CORE_SET_PAGE_LOADING', false);
-          store.dispatch('CORE_SET_ERROR', null);
-        },
-        (error) => {
-          store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
-          store.dispatch('CORE_SET_PAGE_LOADING', false);
-        }
-      );
     },
     (error) => {
       store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
       store.dispatch('CORE_SET_PAGE_LOADING', false);
-    });
+      document.title = _errorTitle();
+    }
+  );
 }
 
 
-function showExploreTopic(store, channelId, id) {
+function showExploreTopic(store, channelId, id, isRoot = false) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
-  store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_TOPIC);
-  store.dispatch('SET_CURRENT_CHANNEL', channelId);
-  cookiejs.set('currentChannel', channelId);
+  if (isRoot) {
+    store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_CHANNEL);
+  } else {
+    store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_TOPIC);
+  }
 
-  const attributesPromise = ContentNodeResource.getModel(id).fetch();
+  const topicPromise = ContentNodeResource.getModel(id).fetch();
   const childrenPromise = ContentNodeResource.getCollection({ parent: id }).fetch();
-  _updateChannelList(store);
-  ConditionalPromise.all([attributesPromise, childrenPromise]).only(
+  const channelsPromise = ChannelResource.getCollection({}).fetch();
+  ConditionalPromise.all([topicPromise, childrenPromise, channelsPromise]).only(
     samePageCheckGenerator(store),
-    ([attributes, children]) => {
-      const pageState = { id };
-      pageState.topic = _topicState(attributes);
+    ([topic, children, channelsData]) => {
+      if (!_handleChannels(store, channelId, _channelListState(channelsData))) {
+        return;
+      }
+      const pageState = {};
+      pageState.topic = _topicState(topic);
       const collection = _collectionState(children);
       pageState.subtopics = collection.topics;
       pageState.contents = collection.contents;
       store.dispatch('SET_PAGE_STATE', pageState);
       store.dispatch('CORE_SET_PAGE_LOADING', false);
       store.dispatch('CORE_SET_ERROR', null);
+      if (isRoot) {
+        const currentChannel = getters.currentChannel(store.state);
+        document.title = _explorePageTitle(currentChannel.title);
```

Comment:
Yes, tota11y right @rtibbles, annoying and not informative as it could be. It's also the question of order of prefixes, so IMHO the page title should ideally contain: `H1` + `Channel Name` + `Kolibri`. 

Examples:

`<title>Explore - Khan Academy - Kolibri</title>`

![selection_117](https://cloud.githubusercontent.com/assets/1457929/18971687/45c924ae-8696-11e6-99ca-4549c2871a20.png)

**Adding type of content might also be useful**
`<title>Addition and subtraction intro - Topic - Khan Academy - Kolibri</title>`

![selection_118](https://cloud.githubusercontent.com/assets/1457929/18971751/8f2f6efa-8696-11e6-855f-378e0e5dcba3.png)

`<title>Add and subtract pieces of fruit - Video - Khan Academy - Kolibri</title>`

![selection_119](https://cloud.githubusercontent.com/assets/1457929/18971889/1d48d3f2-8697-11e6-95f5-6e417a9a7777.png)

`<title>All Users - Manage - (Facility-Name) - Kolibri</title>`

![selection_120](https://cloud.githubusercontent.com/assets/1457929/18971974/812da820-8697-11e6-8460-9b0f645eae49.png)